base/accumulate.jl

```diff
@@ -33,7 +33,7 @@ function accumulate_pairwise!(op::Op, result::AbstractVector, v::AbstractVector)
 end
 
 function accumulate_pairwise(op, v::AbstractVector{T}) where T
-    out = similar(v, promote_op(op, T, T))
+    out = similar(v, promote_op(op, T, T; throw_error=true))
```

Comment:
I'm only using `throw_error=true` in this file since that was the source of the original issue. There may be other cases where this would be useful.